src/enforcer.js

```diff
@@ -173,43 +222,63 @@ export class TrustedTypesEnforcer {
       return;
     }
 
-    if ('ShadowRoot' in window) {
-      this.wrapSetter_(ShadowRoot.prototype, 'innerHTML',
+    if ('ShadowRoot' in this.windowObject_) {
+      this.wrapSetter_(this.windowObject_.ShadowRoot.prototype, 'innerHTML',
           TrustedTypes.TrustedHTML);
     }
     stringifyForRangeHack = (function(doc) {
+      if (!doc.createRange) return false;
       const r = doc.createRange();
       // In IE 11 Range.createContextualFragment doesn't stringify its argument.
       const f = r.createContextualFragment(/** @type {string} */ (
         {toString: () => '<div></div>'}));
       return f.childNodes.length == 0;
-    })(document);
+    })(this.windowObject_.document);
 
-    this.wrapWithEnforceFunction_(Range.prototype, 'createContextualFragment',
-        TrustedTypes.TrustedHTML, 0);
+    if (this.windowObject_.Range) {
+      this.wrapWithEnforceFunction_(
+          this.windowObject_.Range.prototype,
+          'createContextualFragment',
+          TrustedTypes.TrustedHTML, 0);
+    }
 
-    this.wrapWithEnforceFunction_(insertAdjacentObject,
-        'insertAdjacentHTML',
-        TrustedTypes.TrustedHTML, 1);
+    if (this.insertAdjacentObjectPrototype) {
+      this.wrapWithEnforceFunction_(this.insertAdjacentObjectPrototype,
+          'insertAdjacentHTML',
+          TrustedTypes.TrustedHTML, 1);
+    }
 
-    if (getOwnPropertyDescriptor(Document.prototype, 'write')) {
+    if (getOwnPropertyDescriptor(
+        this.windowObject_.Document.prototype, 'write'
+    )) {
       // Chrome
-      this.wrapWithEnforceFunction_(Document.prototype, 'write',
+      this.wrapWithEnforceFunction_(this.windowObject_.Document.prototype,
+          'write',
           TrustedTypes.TrustedHTML, 0);
-    } else {
+    } else if (this.windowObject_.HTMLDocument &&
```

Comment:
You can drop this check, since we assume that `HTMLDocument` is always available.

Reply:
HTMLDocument doesn't exist in domino.